l10n_it_fatturapa_out/models/attachment.py

```diff
@@ -79,6 +79,17 @@ def write(self, vals):
                 )
         return res
 
+    @api.multi
+    def unlink(self):
+        for attachment_out in self:
+            for invoice in attachment_out.out_invoice_ids:
+                invoice_attachments = invoice.fatturapa_doc_attachments
```

Comment:
Se ho ben capito vuoi eliminare gli allegati prodotti da `attachment_out`, puoi fare così:
```suggestion
                invoice.fatturapa_doc_attachments.filtered('is_pdf_invoice_print').unlink()
```
ed eliminare tutto l'`if` qui sotto.

Come ha detto anche @eLBati in https://github.com/OCA/l10n-italy/pull/874/files#r280326295, per com'è adesso elimineresti tutti gli allegati in `fatturapa_doc_attachments` della fattura, anche quelli non prodotti da `attachment_out`.